public/app/features/plugins/admin/components/PluginListItem.tsx

```diff
@@ -23,8 +24,17 @@ function PluginListItemComponent({ plugin, pathName, displayMode = PluginListDis
   const styles = useStyles2(getStyles);
   const isList = displayMode === PluginListDisplayMode.List;
 
+  const reportUserClickInteraction = () => {
+    if (locationService.getSearchObject()?.q) {
```

Comment:
can you use the useLocation hook from react instead?

Reply:
I can, but this way is easier, because when you use useLocation you get an object location, where value of the search is a string like "?filterBy=installed&filterByType=all&q=test" and I would need to parse the string to have only the "q" value

Reply:
(Hopefully soon we will have [useSearchParams()](https://reactrouter.com/en/main/hooks/use-search-params), when we update core to react-router v6!)